text/logs/0150-logging-library-sdk.md

```diff
@@ -0,0 +1,259 @@
+functionality to authors of extensions to language-specific 3rd party logging
+libraries and to end users that want to produce logs in the
+[OpenTelemetry manner](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/overview.md).
+
+The specification defines SDK elements that to some extent mirror the
+OpenTelemetry
+[Trace SDK](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace).
+This ensures uniformity and consistency of the OpenTelemetry specification and
+of the implementations across traces and logs. For additional clarity the
+definitions in this document refer to the Trace analogs where appropriate.
+
+The descriptions of interfaces and methods in this document are intentionally
+very brief. Detailed and precise descriptions will be borrowed from Trace
+specification when this OTEP is submitted as a Log specification PR. I kept the
+descriptions as short as possible in this document to make reviewing it easy and
+fast.
+
+## Specification
+
+Many existing logging libraries have some sort of extension mechanism that
+allows to customize how log records are encoded and delivered to their
+destinations (for example Appender in Log4j or Core in zapcore). The
+OpenTelemetry Logging Library SDK is intended to be used by such extensions to
+emit logs in OpenTelemetry formats.
+
+Note: The functionality that this document proposes will be an SDK package. A
+logging-related API package may be added in the future if we decide to have an
+end-user callable logging API.
+
+We will have the following in the SDK.
+
+### LogEmitterProvider
+
+Methods:
+
+- Get LogEmitter. Accepts the instrumentation library name and version and
+  returns a LogEmitter associated with the instrumentation library.
+- Shutdown.
+- ForceFlush.
+
+LogEmitterProvider can be configured at startup time, to be associated with a
+Resource and with LogProcessor/LogExporter pipeline.
+
+### LogEmitter
+
+Methods:
+
+- Emit(LogRecord). Emits a log record. The LogRecord and the Resource and
+  Instrumentation library associated with the LogEmitter will be converted into
+  a readable LogData and will be pushed through the SDK and the configured
+  LogProcessors and LogExporter. It is expected that the caller will populate
+  trace context related fields (TraceId,SpanId,TraceFlags) if applicable before
+  making the call. Open Question: do we need to also pass the Baggage so that
+  log processors and exporters can use it if they see the need?
+- Flush.
+
+### LogRecord
+
+See LogRecord
+[data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md)
+for the list of fields.
+
+Open Question: should LoggerName field be added to the data model to allow
+logging libraries to supply it? We have an
+[open PR](https://github.com/open-telemetry/opentelemetry-specification/pull/1236).
```

Comment:
OK. thanks for providing the LoggerName PR link here. Please ignore my earlier comment.

Reply:
Please feel free to comment on that PR.